### PR TITLE
Upgrade to `rust-lightning:0.0.117`

### PR DESCRIPTION
**Pull request by luckysori**
Fixes https://github.com/get10101/10101/issues/1651.

This includes an update to `rust-dlc` too.

TODOs:

- [x] Make sure that we haven't dropped any commits from our 10101 dedicated branches in the `rust-dlc` and `rust-lightning` repos. I think we may have dropped some from the `rust-dlc` branch.

### CODE REVIEW

File: crates/ln-dlc-node/src/node/mod.rs [hunk 1]
```diff
@@ -128,6 +132,7 @@ pub struct LiquidityRequest {
 }
 
 /// An LN-DLC node.
+#[derive()]
```
Comment (holzeis) [inline crates/ln-dlc-node/src/node/mod.rs:135]: ❓ Is this doing anything or can that be removed?

File: crates/ln-dlc-node/src/storage.rs [hunk 2]
```diff
@@ -74,100 +40,82 @@ impl Default for TenTenOneInMemoryStorage {
...
-            if !entry.0.contains("monitors") {
-                continue;
+        primary_namespace: &str,
+        _secondary_namespace: &str,
+        key: &str,
+        _lazy: bool,
+    ) -> Result<(), Error> {
+        match key {
+            "manager" => *self.manager.write() = None,
+            "network_graph" => *self.network_graph.write() = None,
```
Comment (holzeis) [inline crates/ln-dlc-node/src/storage.rs:97]: 🔧 Not critical at all, but we should add a handling for the scorer here as well.

**Review by holzeis — APPROVED** [review]

File: crates/ln-dlc-node/src/ln/common_handlers.rs [hunk 3]
```diff
@@ -403,21 +404,29 @@ pub async fn handle_funding_generation_ready<S: TenTenOneStorage, N: Storage>(
...
                 %err,
-                "Cannot open channel due to not being able to create funding tx"
+                "Cannot open channel due to not being able to create funding TX"
             );
+
             node.channel_manager
                 .close_channel(&temporary_channel_id, &counterparty_node_id)
                 .map_err(|e| anyhow!("{e:?}"))?;
 
+            tracing::info!("DID THIS DO ANYTHING?");
```
[the remark this answers is not in the document]
Comment (luckysori) [reply crates/ln-dlc-node/src/ln/common_handlers.rs:414]: Hahaha ooops!

File: crates/ln-dlc-node/src/node/mod.rs (hunk 1 above)
Comment (luckysori) [reply crates/ln-dlc-node/src/node/mod.rs:135]: Nice catch!

**Pull request merged**